### PR TITLE
feat(torch): implement resume mllib option for torchlib: if true, reu…

### PR DESCRIPTION
**Pull request by fantes**
…se previous solver state

### CODE REVIEW

**Review by sileht — APPROVED** [review]

File: docs/api.md [hunk 1]
```diff
@@ -772,6 +772,7 @@ gradient_centralization | bool | yes | false for RANGER, true for RANGER_PLUS| f
 test_interval | int    | yes      | N/A     | Number of iterations between testing phases
 base_lr       | real   | yes      | N/A     | Initial learning rate
 iter_size     | int    | yes      | 1       | Number of passes (iter_size * batch_size) at every iteration
+resume        | bool   | yes      | false   | Whether to resume training from solverstate
```
Comment (BynaryCobweb) [inline docs/api.md:775]: With torch backend it's not called "solverstate" like in caffe

File: src/backends/torch/torchsolver.h [hunk 2]
```diff
@@ -70,6 +70,14 @@ namespace dd
      */
     void save(std::string sfile);
 
+    /**
+     * \brief restore solver state, checks solverstate presecnce  and returns
```
Comment (BynaryCobweb) [inline src/backends/torch/torchsolver.h:74]: presence
\+ reload best metrics

File: src/backends/torch/torchlib.cc [hunk 3]
```diff
@@ -531,9 +531,8 @@ namespace dd
     // create solver
     tsolver.create(_module);
 
-    int it = 0;
-    // reload solver and set it value accordingly
-    it = tsolver.load(this->_mlmodel._sstate, _main_device);
+    int it = tsolver.resume(ad_mllib, this->_mlmodel, _main_device,
```
Comment (BynaryCobweb) [inline src/backends/torch/torchlib.cc:534]: Need to reload `best_to_remove` as well ?

Comment (BynaryCobweb) [reply src/backends/torch/torchlib.cc:534]: Looking at the code it seems that `best_to_remove` is always equal to -1

File: docs/api.md (hunk 1 above)
Comment (fantes) [reply docs/api.md:775]: replaced with "solver state" ;)

File: src/backends/torch/torchsolver.h (hunk 2 above)
Comment (fantes) [reply src/backends/torch/torchsolver.h:74]: ok

File: src/backends/torch/torchlib.cc (hunk 3 above)
Comment (fantes) [reply src/backends/torch/torchlib.cc:534]: > Need to reload `best_to_remove` as well ?

nice one, I did not remember the existence of this one

Comment (fantes) [reply src/backends/torch/torchlib.cc:534]: > Looking at the code it seems that `best_to_remove` is always equal to -1

It should change, it is updated by save_if_best (changed if best)

Comment (BynaryCobweb) [reply src/backends/torch/torchlib.cc:535]: ok I see

**Review by BynaryCobweb — APPROVED** [review]

**Pull request merged**